pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -142,47 +149,97 @@ func (h *NetworkUtilsHandler) UseIptables() bool {
 
 	return true
 }
-func (h *NetworkUtilsHandler) IptablesNewChain(table, chain string) error {
-	iptablesObject, err := iptables.New()
+
+func (h *NetworkUtilsHandler) ConfigureIpv6Forwarding() error {
+	_, err := exec.Command("sysctl", "net.ipv6.conf.all.forwarding=1").CombinedOutput()
+	return err
+}
+
+func (h *NetworkUtilsHandler) Ipv4NatEnabled() bool {
+	lsmod, _ := exec.Command("lsmod").CombinedOutput()
+	lsmodString := string(lsmod)
+	if !strings.Contains(lsmodString, "iptable_nat") && !strings.Contains(lsmodString, "nf_nat_ipv4") {
+		log.Log.Errorf("no ipv4 support")
+		return false
+	}
+	return true
+}
+
+func (h *NetworkUtilsHandler) Ipv6NatEnabled() bool {
+	lsmod, _ := exec.Command("lsmod").CombinedOutput()
+	lsmodString := string(lsmod)
+	if !strings.Contains(lsmodString, "iptable_nat") && !strings.Contains(lsmodString, "nf_nat_ipv6") {
+		log.Log.Errorf("no ipv6 support")
+		return false
+	}
+	return true
+}
+
+func (h *NetworkUtilsHandler) IsIpv6Enabled(link netlink.Link) bool {
+	linkLog := log.Log.With("link", link.Attrs().Name)
+	addrs, err := h.AddrList(link, netlink.FAMILY_V6)
+	if err != nil {
+		linkLog.Reason(err).Errorf("ipv6 disabled, failed retrieving ipv6 addresses")
+		return false
+	}
+	if len(addrs) == 0 {
+		linkLog.Infof("ipv6 disabled, there is no ipv6 addresses")
+	}
+	for _, addr := range addrs {
+		if !addr.IP.IsLinkLocalUnicast() {
```

Comment:
We had to check whether eth0 has only link local address configured because of a mysteries bug that is reproduced in prow only. On some of the ipv4 lanes (not all of them) configuring ipv6 address on the k6t-eth0 bridge causes virt-handler to crash.

This is a workaround till we understand the actual root cause of this crash.

@qinqon can you please elaborate?

Reply:
I think it is sane to not try to configure anything ipv6 related if you are on an ipv4 cluster. Therefore I would suggest to keep a check, but check via the downwards api if you are on an ipv6 enabled cluster.

Reply:
Agree with Roman, is better to rely on PODs interface, at the end is what CNI provides with non link local address or other mechanisms, will add a commit for that.

Reply:
Yeah, link-local addresses can be used for connectivity with the same network.

But if we assume that some CNI relies on link-local, they may as well behave as IPv4 externally fo addressing between pods and use IPv6 link-local only to traverse between host and the pod (extreme example, I know). If we want to allow all CNIs to function, maybe we should forward even link-local through the masquerade, no matter if the network CNI is considered IPv4 or IPv6 externally.

Would that be possible? Can we just setup the binding for all found IPs on the pod interface?

Reply:
The iptables masquerade mechanism is automatically taking the ip address of the outgoing interface. I'm not sure it will take a link local address. If it is not working maybe we can change the rules to use snat instead of masquerade.
Is it ok to finish with this PR and do those experiments in a different one?